src/services/editor.service.ts

```diff
@@ -101,4 +102,50 @@ export class EditorService {
       throw err;
     }
   }
+
+  static async convertToYaml() {
+    try {
+      const yamlContent = FormatService.convertToYaml(this.getValue());
+      yamlContent && this.updateState(yamlContent, true, 'yaml');
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  static async convertToJSON() {
+    try {
+      const jsonContent = FormatService.convertToJSON(this.getValue());
+      jsonContent && this.updateState(jsonContent, true, 'json');
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  static async saveAsYaml() {
+    try {
+      const yamlContent = FormatService.convertToYaml(this.getValue());
+      yamlContent && this.downloadFile(yamlContent, `${this.fileName}.yaml`);
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  static async saveAsJSON() {
+    try {
+      const jsonContent = FormatService.convertToJSON(this.getValue());
+      jsonContent && this.downloadFile(jsonContent, `${this.fileName}.json`);
+    } catch (err) {
+      console.error(err);
+      throw err;
+    }
+  }
+
+  private static fileName = 'asyncapi';
+
+  private static downloadFile(content: string, fileName: string) {
+    return fileDownload(content, fileName);
+  }
```

Comment:
🧐

Reply:
`fileName` is a base name, and then to it is added extension - see functions `saveJSON` etc :) There are any problems with shadowing etc.